tests/python/frontend/onnx/test_forward.py

```diff
@@ -5174,6 +5172,10 @@ def test_onnx_nodes(target, dev, onnx_test):
         # roialign results to 4 decimal places
         atol = 1e-4
 
+    if "to_BFLOAT16" in test_dir:
+        # casting to BFLOAT has a larger accuracy loss
+        atol = 1
```

Comment:
nit: curious what the magnitude of expected outputs was, like on the order of 10s, 100s, etc that you saw in the tests? 

1 does seem pretty high and more than I would expect for a cast op -- I would've expected tvm's vs ort's cast implementation to be pretty similar, even for bfloat16. I'm wondering if you have thoughts as to what might cause that discrepancy?

Reply:
i completely agree, the difference here seems pretty high

here's what I was seeing for the difference in outputs:

```
E       AssertionError: 
E       Not equal to tolerance rtol=1e-05, atol=1e-05
E       
E       Mismatched elements: 5 / 12 (41.7%)
E       Max absolute difference: 65535
E       Max relative difference: 4.10620301
E        x: array([[16117, 16117, 16127, 16209],
E              [16112, 16209, 15959, 16185],
E              [32704, 32640, 32640, 65408]], dtype=uint16)
E        y: array([[16117, 16118, 16128, 16210],
E              [16113, 16209, 15960, 16185],
E              [32704, 32640, 32640, 65408]], dtype=uint16)
```

Reply:
I see, this makes me wonder if we're handling an edge case differently than ORT is. 

also curious if these differences still occur if lower magnitude inputs are supplied, since any differences are significant at lower magnitudes.

Reply:
The errors might be large since you are comparing things in uint16 space which might not be indicative. 

E.g. 0 (e.g. 0x0000) and -0 (e.g. 0x1000) would differ by a lot in uint space but not at all in bfloat state
 I would cast back to bfloat16 compatible state

E.g. https://en.wikipedia.org/wiki/Half-precision_floating-point_format
Which gives the float64 number representing the sameish number as a bfloat16 cast. It does not handle special values though

Reply:
Difference might be due to edge case handling yeah, you can see if `15960` and `32640` are special values of some sort.

Reply:
Oh rereviewing this, i was comparing two dissimilar arguments. Anyway, can you translate these uint16 values to the real numbers they represent under a re-interpret cast to bfloat16? This way you don't need to set the tolerance up by a ton.

Reply:
i was hoping to keep the edge case handling to a minimum since this is a common test function. I understand that reinterpreting to bfloat16 would allow the tolerance to seem lower, but it would require adding a lot more code to `test_onnx_nodes` in order to handle just the single `to_bfloat16` unit test (i.e. I'd have to reinterpret both the expected onnx output and the tvm output to bfloat16) -- would you say we should still have the reinterpret cast?

Reply:
I think it's fine to set atol to 1. But would like a comment on why it's 1 and why this really isn't that big deal (e.g. it's just the mantissa being off by 1 which is the minimum difference between any two floating point numbers)